Sources/SwiftSyntax/SwiftSyntaxCompatibility.swift

```diff
@@ -28,3 +28,175 @@ public typealias AccessPathComponentSyntax = ImportPathComponentSyntax
+      unexpectedBeforeGenericParameters,
+      genericParameterClause: genericParameters,
+      unexpectedBetweenGenericParametersAndBaseType,
+      baseType: baseType,
+      unexpectedAfterBaseType,
+      arena: arena
+    )
+  }
+}
+
+public extension RawEnumDeclSyntax {
+  @available(*, deprecated, renamed: "unexpectedBetweenIdentifierAndGenericParameterClause")
+  var unexpectedBetweenIdentifierAndGenericParameters: RawUnexpectedNodesSyntax? { unexpectedBetweenIdentifierAndGenericParameterClause }
+
+  @available(*, deprecated, renamed: "genericParameterClause")
+  var genericParameters: RawGenericParameterClauseSyntax? { genericParameterClause }
+
+  @available(*, deprecated, renamed: "unexpectedBetweenGenericParameterClauseAndInheritanceClause")
+  var unexpectedBetweenGenericParametersAndInheritanceClause: RawUnexpectedNodesSyntax? { unexpectedBetweenGenericParameterClauseAndInheritanceClause }
+
+  @available(*, deprecated, message: "Use an initializer with a genericParameterClause argument.")
+  init(
+    _ unexpectedBeforeAttributes: RawUnexpectedNodesSyntax? = nil,
+    attributes: RawAttributeListSyntax?,
+    _ unexpectedBetweenAttributesAndModifiers: RawUnexpectedNodesSyntax? = nil,
+    modifiers: RawModifierListSyntax?,
+    _ unexpectedBetweenModifiersAndEnumKeyword: RawUnexpectedNodesSyntax? = nil,
+    enumKeyword: RawTokenSyntax,
+    _ unexpectedBetweenEnumKeywordAndIdentifier: RawUnexpectedNodesSyntax? = nil,
+    identifier: RawTokenSyntax,
+    _ unexpectedBetweenIdentifierAndGenericParameters: RawUnexpectedNodesSyntax? = nil,
+    genericParameters: RawGenericParameterClauseSyntax?,
+    _ unexpectedBetweenGenericParametersAndInheritanceClause: RawUnexpectedNodesSyntax? = nil,
+    inheritanceClause: RawTypeInheritanceClauseSyntax?,
+    _ unexpectedBetweenInheritanceClauseAndGenericWhereClause: RawUnexpectedNodesSyntax? = nil,
+    genericWhereClause: RawGenericWhereClauseSyntax?,
+    _ unexpectedBetweenGenericWhereClauseAndMemberBlock: RawUnexpectedNodesSyntax? = nil,
+    memberBlock: RawMemberDeclBlockSyntax,
+    _ unexpectedAfterMemberBlock: RawUnexpectedNodesSyntax? = nil,
+    arena: __shared SyntaxArena
+  ) {
+    self.init(
+      unexpectedBeforeAttributes,
+      attributes: attributes,
+      unexpectedBetweenAttributesAndModifiers,
+      modifiers: modifiers,
+      unexpectedBetweenModifiersAndEnumKeyword,
+      enumKeyword: enumKeyword,
+      unexpectedBetweenEnumKeywordAndIdentifier,
+      identifier: identifier,
+      unexpectedBetweenIdentifierAndGenericParameters,
+      genericParameterClause: genericParameters,
+      unexpectedBetweenGenericParametersAndInheritanceClause,
+      inheritanceClause: inheritanceClause,
+      unexpectedBetweenInheritanceClauseAndGenericWhereClause,
+      genericWhereClause: genericWhereClause,
+      unexpectedBetweenGenericWhereClauseAndMemberBlock,
+      memberBlock: memberBlock,
+      unexpectedAfterMemberBlock,
+      arena: arena
+    )
+  }
+}
```

Comment:
The `Raw` don’t need to be stable. I’m adding some documentation about that in https://github.com/apple/swift-syntax/pull/1696

Reply:
I have read the SPI.md document and removed the code intended for preserving the Raw type initializer 🙂